core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -1278,11 +1287,22 @@ private[spark] object SparkSubmitUtils {
       remoteRepos: Option[String],
       ivyPath: Option[String]): IvySettings = {
     val file = new File(settingsFile)
-    require(file.exists(), s"Ivy settings file $file does not exist")
-    require(file.isFile(), s"Ivy settings file $file is not a normal file")
```

Comment:
@shardulm94, does it work if you set `spark.jars.ivySettings` to `./ivysettings.xml` and pass `ivysettings.xml` to `spark.yarn.files`?

Reply:
I know you might have to copy `ivysettings.xml` to the current working directory when Spark submit runs but I think it might work for your usecase.

Reply:
Yeah, as you pointed out it only works if I have the copy in the current working directory. However we cannot which directory our users launch spark-submit from.

What do you think of this? In Yarn Client, we can add the ivySettings file to `spark.yarn.dist.files` or `__spark__conf__.zip` and then we can modify the property `spark.jars.ivySettings` to change it to `./ivysettings.xml` or `__spark__conf__/ivysettings.xml`.

Reply:
Adding it into `__spark__conf__` sounds okay to me ... but I would prefer to have second opinions from Yarn experts such as @tgravescs, @mridulm or @jerryshao.

Reply:
I tried this out, and it looks like it can be handled pretty cleanly this way targeting just YARN. https://github.com/shardulm94/spark/pull/1/commits/12709f0def5a20a0448e602d3ac8216396c8dea3

Reply:
I updated the PR with this approach.